buffer/src/main/java/io/netty/buffer/UnsafeByteBufUtil.java

```diff
@@ -544,23 +544,23 @@ static void setBytes(AbstractByteBuf buf, long addr, int index, byte[] src, int
         }
     }
 
-    static void setBytes(AbstractByteBuf buf, long addr, int index, ByteBuffer src) {
+    static void setBytes(AbstractByteBuf buf, MemoryAddress addr, int index, ByteBuffer src) {
         final int length = src.remaining();
         if (length == 0) {
             return;
         }
 
-        if (src.isDirect()) {
-            buf.checkIndex(index, length);
-            // Copy from direct memory
-            long srcAddress = PlatformDependent.directBufferAddress(src);
-            PlatformDependent.copyMemory(srcAddress + src.position(), addr, length);
-            src.position(src.position() + length);
-        } else if (src.hasArray()) {
+        if (src.hasArray()) {
```

Comment:
the if, else has been changes around? Is there a reason?

Reply:
Because the `if (src.hasArray())` branch does not have to perform a `MemoryAddress` add, I reasoned it would likely now be more efficient that the `isDirect` branch - but this probably renders the `isDirect` branch unnecessary

Reply:
I could revert to the original pattern, or delete the `isDirect` branch - which do you reckon?